src/main/java/com/example/Employee_Management_System/Service/SalaryService.java

```diff
@@ -53,13 +55,19 @@ public Salary update(SalaryRequest request) {
     }
 
 
-    public Salary getById(int id){
-        return salaryRepository.findById(id).get();
+    public Salary getById(int id) {
+        return salaryRepository.findById(id)
+                .orElseThrow(() -> new NotFoundException(String.format("Salary not found by id: %s", id)));
     }
 
-    public Employee maxSalary() {
-        Salary salary = salaryRepository.maxSalary();
-        return employeeRepository.findById(salary.getEmpid()).get();
+    public List<Employee> maxSalary() {
+        List<Salary> salaries = salaryRepository.maxSalary();
+        List<Employee> employees = new ArrayList<Employee>();
+
+        for (Salary salary : salaries) {
+            employees.add(employeeRepository.findById(salary.getEmpid()).get());
```

Comment:
optimize the call in loop using sql IN operator to get all the employees info in one shot